src/main/java/sandbox/JdbcSqlIndexIterator.java

```diff
@@ -0,0 +1,82 @@
+
+package sandbox;
+
+import org.javarosa.core.services.storage.IStorageIterator;
+import org.javarosa.core.services.storage.Persistable;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+/**
+ * Iterator that only processes id indexes, and not real data.
+ * 
+ * @author ctsims
+ */
+public class JdbcSqlIndexIterator<E extends Persistable> extends JdbcSqlStorageIterator<E> {
```

Comment:
Instead of extending this, these two should be either merged, or use a common ancestor, this was just the fastest way to bootstrap things